### ISSUE TITLE
Maven in basic Eclipse Java package

### ISSUE THREAD

**Issue by elharo** [GoogleCloudPlatform/google-cloud-eclipse#715]
Per JavaOne talk, Maven integration should (in 4.6) now be in the basic non-Java EE package. Figure out what this means for our docs and plug-ins. (and file a bug with eclipse if this is not true)

http://www.eclipse.org/downloads/packages/eclipse-ide-java-developers/neonr

**Comment by briandealwis** [comment]
But it doesn't have WTP.

**Comment by chanseokoh** [comment]
The Mars version of the non-Java EE package (Eclipse IDE for Java Developers) also has the _Maven Integration for Eclipse_: http://www.eclipse.org/downloads/packages/eclipse-ide-java-developers/mars2. In fact, I have been using the non-Java EE Eclipse Mars since the beginning, and I didn't have to install M2E on it.

Also, from my experience, all of our CT4E features work as expected, and this even includes local runs (although the shortcut context menu 'Run As' > 'Run on Server' is missing).

About WTP and local runs:
@akerekes observed in one of our previous UX sessions that at least some of WTP parts are automatically pulled in at the time of installing our plugin. I think this is true, because I verified that initially you don't have the Servers, but you can open the view after installing our plugin (by Windows > Show View > Others...).

Then, you can create our App Engine server (in a stopped state), select a module, and run or debug, following these general instructions: http://crunchify.com/step-by-step-guide-to-setup-and-install-apache-tomcat-server-in-eclipse-development-environment-ide/

Moreover, if a user installs WTP (i.e., Eclipse Java EE Developer Tools) from the Marketplace, then it adds the shortcut menu items 'Run on Server' and 'Debug on Server'.

So, basically, I think we could support non-Java EE packages although this needs to be tested thoroughly.

**Comment by briandealwis** [comment]
We should maybe add some of the WTP features to our feature's required-features.

**Comment by elharo** [comment]
docs updated

**Comment by chanseokoh** [comment]
Just as a record:

> So, basically, I think we could support non-Java EE packages although this needs to be tested thoroughly.

It seems like the non-Java EE lacks more features that we need. Deploying a Maven-based project seems to fail.

**Comment by briandealwis** [comment]
I wonder if we should add some feature dependencies to ensure the functionality is there.
